src/lib.rs

```diff
@@ -51,7 +58,17 @@ pub async fn readable(url: Uri) -> Result<impl IntoResponse, (StatusCode, Html<S
         )
     })?;
 
-    let body = reqwest::get(url.clone())
+    let forwarded_agent = match ua_header {
+        Some(TypedHeader(ua)) => {
+            ReqwestHeaderValue::from_str(ua.as_str()).unwrap_or(DEFAULT_USER_AGENT)
+        },
+        None => DEFAULT_USER_AGENT,
+    };
+
+    let client = reqwest::Client::new();
+    let body = client.get(url.clone())
+        .header(USER_AGENT, forwarded_agent)
```

Comment:
You _could_ move the header the detection into a separate function (maybe even in a separate `utils` module, including the default user agent). This way you could just call `forwarded_agent(ua_header)` here. Maybe a little cleaner and reusable. Definitely a minor thing, though. 😉

Reply:
That's definitely better. Gets rid of the conflict between `axum::http::HeaderValue` and `reqwest::header::HeaderValue` as well.

Reply:
Perfect, thanks!